app/controllers/spree/admin/images_controller.rb

```diff
@@ -35,34 +35,24 @@ def create
 
         @object.attributes = permitted_resource_params
 
-        if @object.save
-          flash[:success] = flash_message_for(@object, :successfully_created)
-          redirect_to location_after_save
-        else
-          respond_with(@object)
-        end
-      rescue ActiveStorage::IntegrityError
-        @object.errors.add :attachment, :integrity_error
-        respond_with(@object)
+        @object.save!
+        flash[:success] = flash_message_for(@object, :successfully_created)
+
+        redirect_to location_after_save
+      rescue ActiveRecord::RecordInvalid => e
+        respond_with_error(e)
       end
 
       def update
         @url_filters = ::ProductFilters.new.extract(request.query_parameters)
         set_viewable
 
-        if @object.update(permitted_resource_params)
-          flash[:success] = flash_message_for(@object, :successfully_updated)
+        @object.update!(permitted_resource_params)
+        flash[:success] = flash_message_for(@object, :successfully_updated)
 
-          respond_with do |format|
-            format.html { redirect_to location_after_save }
-            format.turbo_stream
```

Comment:
`±` It looks like we lose the `turbo_stream` format for `update` success. This was a shortcut which allows the image to be updated on the page without reloading the page (and losing scroll position). Can you please bring that back?

Reply:
I can see that will make the turbo request spec fail. I can't think of a clean way to re-use the shared_examples, so you might need to just copy them, or add parameters to shared_examples.

Reply:
Hello @dacook , I made a blunder by removing the classical html view. So, I quickly re-added them, in a new method to factor code between `create` and `edit` methods. I made 2 commits spaced by 45mn. The last one is green. So, I guess your comment came after the first commit of the two of them.

Reply:
Yes, I see your last commit brings the `html` format back for **_error_** case 👍 
But, the `turbo_stream` format for the update **_success_** case is still missing. 

You can test this by scrolling to the product at the bottom of the page, and updating the image. It should update without jumping back to the top of the page.

(as a side note, turbo_stream format is also missing for `create` success, but it was already missing before this PR)

Reply:
Hello @dacook , you are right.
I set the turbo responses both for `update` & `create`.
It also had a small side effect on the request spec.